xdsl/dialects/snitch_runtime.py

```diff
@@ -45,6 +46,8 @@ class SnitchRuntimeGetInfo(SnitchRuntimeBaseOperation, ABC):
 
     result: OpResult = result_def(i32)
 
+    traits = frozenset([NoMemoryEffect()])
+
```

Comment:
(Same comment for all snitch ops)
They seem strange to me. Given that those operation have no operands, `NoMemoryEffect` implies that the result will always be the same, i.e., you could replace those op by a given `arith.constant`. Is that the case?

Reply:
Ah, nevermind, I'm overinterpreting the thing; it can be context-dependent, like `gpu.thread_id` for example. So LGTM as long as it sounds right to for this dialect!